### PR TITLE
Upgrade to Node 18 and remove playroom package

### PR DESCRIPTION
**Pull request by cashd**
- update CI jobs to node 18
- Added icons folder to prettierignore to avoid prettier no parser found errors for svg files. We can find a parser in a follow up PR.
- updated yarn to 3.6.0
- update dependencies in docs packages that rely on older versions of Node, particularly webpack.
- lint fixes for components/*
- enforce node 18 runtime in docs
- removed playroom package and references in the docs package

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -35,14 +35,16 @@
     "@types/glob": "^7.2.0",
     "@types/lodash": "^4.14.176",
     "@types/mdx-js__react": "^1.5.5",
-    "@types/react": "^17.0.33",
+    "@types/react": "^18.0.1",
     "@types/react-dom": "^17.0.10",
     "@types/rimraf": "^3.0.2",
-    "@vanilla-extract/babel-plugin": "^1.1.2",
```
Comment (cashd) [inline docs/package.json:41]: https://github.com/vanilla-extract-css/vanilla-extract/releases?q=next&expanded=true

File: .prettierignore [hunk 2]
```diff
@@ -7,3 +7,6 @@ components/src/components/icons/generated/
 
 .yarn/*
 yarn.lock
+
+icons
+*.svg
```
Comment (nathanielgrossman) [inline .prettierignore:12]: why are these being added to the prettierignore?

**Review by nathanielgrossman — COMMENTED** [review]
Could you explain the changes to icons/prettier?

File: .prettierignore (hunk 2 above)
Comment (cashd) [reply .prettierignore:12]: I was getting an error in the commit pre-hook from prettier. Something like `no parser found for svg files`

**Comment by cashd** [comment]
> Could you explain the changes to icons/prettier?

sorry I had a changelog but lost the progress. just added one to the PR description now

**Comment by cashd** [comment]
I just reset the change to the icons dir

File: docs/package.json [hunk 3]
```diff
@@ -35,14 +35,16 @@
...
     "@vanilla-extract/css": "^1.6.3",
-    "@vanilla-extract/next-plugin": "^1.0.1",
+    "@vanilla-extract/next-plugin": "^2.1.1",
```
[the remark this answers is not in the document]
Comment (cashd) [reply docs/package.json:42]: yes the minimum version is 12.0.5

File: .prettierignore (hunk 2 above)
Comment (tvler) [reply .prettierignore:12]: might just wanna rm this for now and decrease the scope since the rest of the icon stuff was reverted

**Review by nathanielgrossman — APPROVED** [review]

**Review by tvler — APPROVED** [review]

**Pull request merged**